Userland/Libraries/LibWeb/DOM/Event.cpp

```diff
@@ -18,9 +18,13 @@ namespace Web::DOM {
 
 JS_DEFINE_ALLOCATOR(Event);
 
+// https://dom.spec.whatwg.org/#concept-event-create
 JS::NonnullGCPtr<Event> Event::create(JS::Realm& realm, FlyString const& event_name, EventInit const& event_init)
```

Comment:
Because this is called from Event::construct_impl, that means that this change is not quite correct.

Event::construct_impl is called by the generated EventConstructor code when javascript creates an event. That means that with your change here, all events, even synthetic events created by JS, will be trusted.

Reply:
I think the solution here is to have construct_impl call heap().allocate itself, so that C++ code / specifications can call Event::create(), and leave construct_impl for the bindings code.

Reply:
> I think the solution here is to have construct_impl call heap().allocate itself, so that C++ code / specifications can call Event::create(), and leave construct_impl for the bindings code.

OK, thanks much — made it so (amended that change to the to commit, and force-pushed here to the PR branch)

Reply:
> (and a test that synthetic events are not trusted :) )

It seems like that may already be getting tested by the existing WPT test here:

https://github.com/web-platform-tests/wpt/blob/e74098ed10aa0a4fc6ad6b8d6e354895addd00d1/dom/events/Event-initEvent.html#L12-L20

Does that seem right? If so, should I also add an additional test in-tree?

Reply:
> If so, should I also add an additional test in-tree?

IMO we should always be adding in-tree tests. Out-of-tree tests like WPT and test262 are great, but aren't practical for providing immediate local feedback if a change we make broke something.

Reply:
> IMO we should always be adding in-tree tests. Out-of-tree tests like WPT and test262 are great, but aren't practical for providing immediate local feedback if a change we make broke something.

Yep, makes sense — I’ll get one added

Reply:
OK, amended a test to the commit